cache/image_cache.go

```diff
@@ -46,6 +47,10 @@ func NewImageCacheFromName(name string, keychain authn.Keychain) (*ImageCache, e
 	if err != nil {
 		return nil, fmt.Errorf("creating new cache image %q: %v", name, err)
 	}
+	if err := emptyImage.SetOS(runtime.GOOS); err != nil {
+		return nil, fmt.Errorf("setting OS new cache image %q: %w", name, err)
+	}
```

Comment:
I am going to assume we need to do something like this as well?

https://github.com/buildpacks/pack/blob/main/internal/buildpackage/builder.go#L109

Reply:
Yeah, exactly. I was just checking to see if this had any adverse side effects in CI, since I wasn't seeing any failures locally, despite the images being invalid.

Reply:
I've now added the shim layer and CI is still passing. I intend to rework this a bit more add more testing (also try to re-enable Windows acceptance tests that use this). 

I'm still not sure whether adding the shim-layer should be moved into `imgutil`. The `image.AddLayer(tarPath)` interface kinda goes against adding layers any other way, plus one of `pack`'s `WorkableImage` implementations doesn't use an `imgutil` image so would still need to add the shim layer anyways.

Did you have any thoughts on a cleaner interface for now?

Reply:
1. I think an `AddLayerFromReader` would make sense as an imgutil feature.
1. It would also make sense in `imgutil` added the shim layer when creating an empty windows image.
1. I don't really understand why we added an implementation for image layouts directly to pack instead of adding it to `imgutil`, but I don't have all the context on that one.